engine/standalone/util/src/main/scala/pl/touk/nussknacker/engine/standalone/utils/JsonSchemaStandaloneSourceFactory.scala

```diff
@@ -0,0 +1,91 @@
+package pl.touk.nussknacker.engine.standalone.utils
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.Json
+import org.everit.json.schema.loader.SchemaLoader
+import org.everit.json.schema.Validator
+import org.everit.json.schema.Schema
+import org.json.JSONObject
+import pl.touk.nussknacker.engine.api.process.SourceTestSupport
+import pl.touk.nussknacker.engine.api.test.{NewLineSplittedTestDataParser, TestDataParser}
+import pl.touk.nussknacker.engine.api.typed.{TypedMap, _}
+import pl.touk.nussknacker.engine.api.{CirceUtil, MetaData, MethodToInvoke, ParamName}
+import pl.touk.nussknacker.engine.standalone.api.openapi.OpenApiSourceDefinition
+import pl.touk.nussknacker.engine.standalone.api.types.GenericResultType
+import pl.touk.nussknacker.engine.standalone.api.{ResponseEncoder, StandalonePostSource, StandaloneSourceFactory}
+import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
+import pl.touk.nussknacker.engine.util.typing.{JsonToTypedMapConverter, SchemaToTypingResultConverter}
+
+import java.nio.charset.StandardCharsets
+
+class JsonSchemaStandaloneSourceFactory extends StandaloneSourceFactory[TypedMap] {
+
+  private val jsonEncoder = BestEffortJsonEncoder(failOnUnkown = true, getClass.getClassLoader)
+
+  @MethodToInvoke
+  def create(@ParamName("schema") schemaStr: String)(implicit metaData: MetaData) : StandalonePostSource[TypedMap] =
+    new JsonSchemaStandaloneSource(schemaStr, metaData, jsonEncoder)
+
+  override def clazz: Class[_] = classOf[TypedMap]
+
+}
+
+class JsonSchemaStandaloneSource(schemaStr: String, metaData: MetaData, jsonEncoder: BestEffortJsonEncoder) extends StandalonePostSource[TypedMap] with LazyLogging with ReturningType with SourceTestSupport[TypedMap] {
+  protected val validator: Validator = Validator.builder().build()
+  protected def prepareSchema(rawSchema: JSONObject) = {
+    SchemaLoader.builder()
+      .useDefaults(true)
+      .schemaJson(rawSchema)
+      .draftV7Support()
+      .build().load().build()
+      .asInstanceOf[Schema]
+  }
+
+  private val rawSchema: JSONObject = new JSONObject(schemaStr)
+  private val schema: Schema = prepareSchema(rawSchema)
+
+  override def parse(parameters: Array[Byte]): TypedMap = {
+    val parametersString = new String(parameters, StandardCharsets.UTF_8)
+    validateAndReturnTypedMap(parametersString)
+  }
+
+  override def openApiDefinition: Option[OpenApiSourceDefinition] = {
+    val json = decodeJsonWithError(schemaStr)
+    val properties = metaData.additionalFields.map(_.properties).getOrElse(Map.empty)
```

Comment:
are we sure all properties should be in external definition?

Reply:
True, I moved part of generating description to `protected val openApiDescription` which you can override if you will.